lib/cocoapods-core/podfile/target_definition.rb

```diff
@@ -907,9 +905,18 @@ def podspec_dependencies
         podspecs.map do |options|
           file = podspec_path_from_options(options)
           spec = Specification.from_file(file)
-          all_specs = [spec, *spec.recursive_subspecs]
-          all_deps = all_specs.map { |s| s.dependencies(platform) }.flatten
-          all_deps.reject { |dep| dep.root_name == spec.root.name }
+          subspec_names = options[:subspecs] || options[:subspec]
+          specs = if subspec_names.blank?
```

Comment:
Could this raise an error if the type is incorrect? Like if we receive `:subspecs => [1234]`

Reply:
Is it strange that each parameter determines the type?

Reply:
Well they sort of have a type already, even if its implicit rather than explicit. Passing a `Hash` or something else into `:subspecs` doesn't really make sense

Reply:
add `subspec_names.map!(&:to_s)` ?

Reply:
It seams that it will not raise an error if the `subspecs => [1234]`.
```
          subspec_names = options[:subspecs] || options[:subspec]
          specs = if subspec_names.blank?
                    [spec]
                  else
                    subspec_names = [subspec_names] if subspec_names.is_a?(String)
                    subspec_names.map { |subspec_name| spec.subspec_by_name("#{spec.name}/#{subspec_name}") }
                  end
```
It will raise an error if `:subspec=>123`.

Reply:
Yeah, I'd rather not just convert everything to a String since that becomes an expected functionality of the API

It makes sense that if you are specifying a subspec name, it should be a String.

Reply:
I add the type checking in the `store_podspec` method.